packages/expo-av/src/WebFullscreenUtils.ts

```diff
@@ -0,0 +1,105 @@
+const supportsFullscreenAPI = (element: HTMLMediaElement): boolean =>
+  'requestFullscreen' in element;
+
+const supportsWebkitFullscreenAPI = (element: HTMLMediaElement): boolean =>
+  'webkitEnterFullScreen' in element;
+
+const supportsWebkitFullscreenChangeEvent = (): boolean => {
+  const element = document.createElement('video');
+  element.setAttribute('onwebkitfullscreenchange', 'return;');
+  return typeof element['onwebkitfullscreenchange'] === 'function';
+};
+
+const supportsMsFullscreenAPI = (element: HTMLMediaElement): boolean =>
+  'msRequestFullscreen' in element;
+
+/**
+ * A helper that adds an event listener to an element. The key value-add over
+ * the native addEventListener is that it returns a function that will remove
+ * the event listener. This allows the setup and teardown logic for a listener
+ * to be easily colocated.
+ */
+function addEventListener(
+  element: Document | HTMLElement,
+  eventName: string,
+  listener: EventListenerOrEventListenerObject
+): () => any {
+  element.addEventListener(eventName, listener);
+  return () => element.removeEventListener(eventName, listener);
+}
+
+/**
+ * Switches a video element into fullscreen.
+ */
+export async function requestFullscreen(element: HTMLMediaElement): Promise<void> {
+  if (supportsFullscreenAPI(element)) {
+    return element.requestFullscreen();
+  } else if (supportsWebkitFullscreenAPI(element)) {
+    element['webkitEnterFullScreen']?.();
+    return Promise.resolve();
+  } else if (supportsMsFullscreenAPI(element)) {
+    element['msRequestFullscreen']?.();
+    return Promise.resolve();
+  } else {
+    return Promise.reject(new Error('Fullscreen not supported'));
```

Comment:
This is already an async function so no need for Promise.reject

```suggestion
    throw new Error('Fullscreen not supported');
```